compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -4020,6 +4020,13 @@ private boolean isXmlNamespaceAttribute(BLangXMLAttribute attribute) {
 
     public void visit(BLangXMLTextLiteral bLangXMLTextLiteral) {
         checkStringTemplateExprs(bLangXMLTextLiteral.textFragments, false);
+        List<BLangExpression> literalValues = bLangXMLTextLiteral.textFragments;
+        if (literalValues.get(0).getKind() == NodeKind.LITERAL &&
+                ((String) ((BLangLiteral) literalValues.get(0)).value).isEmpty()) {
```

Comment:
Don't we also have to check if ` literalValues.size() == 1`?

The following also seems to be allowed atm

```ballerina
public function main() {
    string x = "foo";
    xml<never> b = xml `${""} ${x}`;
}
```